### ISSUE TITLE
does not work with java 9 and beyond

### ISSUE THREAD

**Issue by terefang** [infrastructor/infrastructor#124]
output from "infrastructor run -l 4 -f src/main/scripts/install.groovy" under java 9+

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.vmplugin.v7.Java7$1 (file:/data/n500579/sdkman/candidates/infrastructor/0.2.2/lib/groovy-2.5.4.jar) to constructor java.lang.invoke.MethodHandles$Lookup(java.lang.Class,int)
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.vmplugin.v7.Java7$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[DEBUG] Application variables: [:]
[DEBUG] Application profile: null
[DEBUG] loading application settings from default file '/data/n500579/sdkman/candidates/infrastructor/current/conf/settings.groovy'
[DEBUG] loading system settings from file '/home/n500579/.infrastructor/settings.groovy'
[DEBUG] no system settings file found /home/n500579/.infrastructor/settings.groovy
[DEBUG] System settings: [:]
[DEBUG] Effective settings: [:]
[INFO] running file: 'src/main/scripts/install.groovy'
```

output from "infrastructor run -l 3 -f src/main/scripts/install.groovy" under java 8

```
[DEBUG] Application variables: [:]
[DEBUG] Application profile: null
[DEBUG] loading application settings from default file '.../sdkman/candidates/infrastructor/current/conf/settings.groovy'
[DEBUG] loading system settings from file '.../.infrastructor/settings.groovy'
[DEBUG] no system settings file found .../.infrastructor/settings.groovy
[DEBUG] System settings: [:]
[DEBUG] Effective settings: [:]
[INFO] running file: 'src/main/scripts/install.groovy'
[INFO] task: 'test directory'
[DEBUG] running task: test directory with parallelism: 1 on nodes: [null]
[DEBUG] action: ShellAction, node: null, params: [user:null, sudopass:null, command:test -d /opt/fm ; echo $?, failfast:true]
[DEBUG] connecting to node: null, host: 127.1.192.130, port: 60022
[DEBUG] ssh: test -d /opt/fm ; echo $?
[FILE] 'src/main/scripts/install.groovy'
[TASK] 'test directory'
[NODES] WAITING: 0 RUNNING: 1 FAILED: 0 DONE: 0
     0% <------------------------------------->
[DEBUG] disconnecting from node: null, host: 127.1.192.130, port: 60022
[INFO] task: 'test directory', done on 1 node|s
[FILE] 'src/main/scripts/install.groovy'
[INFO] file: 'src/main/scripts/install.groovy' is done

EXECUTION COMPLETE in 1.489 seconds
```

**Comment by tiurikov** [comment]
Infrastructor depends on the latest stable (at this moment) Groovy Runtime 2.5.x which does not support Java 9+.
Groovy 3.0 should support Java 9+, but it is not stable yet (3.0.0-alpha-4). 
Once it goes GA I will update dependencies on Infrastructor's side.

**Comment by tiurikov** [comment]
an experimental fix #125

**Comment by tiurikov** [comment]
Fixed in 0.3.0
